scripts/update-l10n-sources.js

```diff
@@ -0,0 +1,33 @@
+//@ts-check
+
+/**
+ * This is a pre-hook task that checks if the files `docusaurus.config.js` or
+ * `sidebars.js` have been modified and if so makes sure that the files
+ * needed for localization are regenerated.
+ */
+
+const { getChanges } = require('./utils/git-commands');
+const del = require('del');
+const { execute } = require('./utils/execute');
+
+const files = ['docusaurus.config.js', 'sidebars.js'];
+
+const start = async () => {
+  console.log(`Checking if the following files have been modified:
+${files.join('\n')}`);
+  const output = await getChanges();
+
+  const needsRegeneration = files.find((file) => {
+    return output.includes(`M ${file}`);
+  });
```

Comment:
nit: `some` is a slightly better function here since we're using `needsRegeneration` as a boolean value.

```suggestion
  const needsRegeneration = files.some((file) => {
    return output.includes(`M ${file}`);
  });
```